api/src/main/java/net/kyori/adventure/text/Component.java

```diff
@@ -47,7 +47,37 @@
 import org.checkerframework.checker.nullness.qual.Nullable;
 
 /**
- * A component.
+ * A component represents text that can be displayed to the Minecraft client through different mediums.
+ *
+ * <p>Some rules apply to all types of components:</p>
+ * <ul>
+ *
+ *   <li>
+ *     Components are <i>immutable</i> after they are created.
+ *
+ *   <p>This means that:</p>
+ *      <pre>
+ *      myComponent.style(style)
+ *      </pre>
+ *   will not change "myComponent", but rather return a new component with the content from and
+ *   the styles of the old component in addition to the new style. The same logic applies when
+ *   applying decorations/colors, appending other components and every other operation that
+ *   modifies components after they are created.</li>
+ *
+ *   <li>
+ *     All components can be built using their respective builders which are <i>mutable</i>.
+ *     In addition to creating components from builders, all components can also be turned into builders.
+ *   </li>
+ *
+ *   <li>
+ *     The content of a component can not be directly fetched from the object itself. To get this, you
```

Comment:
This feels a bit vague to me. You can definitely retrieve some sort of "content" from the individual method types, so this could be reworded to say serializers are needed to create various *string representations* of components.

Reply:
They may not be. It could conceivably be converted into a Configurate node tree, for example. The serailisers do allow for that thanks to the generic bound.

Reply:
Yeah -- this wording almost feels like it's saying the opposite of what's actually true -- that `Component` instances are the most direct machine-readablae representation of the contents of a component, but need a serializer to be loaded from the wire/from a configuration file.